src/components/MyDiagramContainer.tsx

```diff
@@ -6,8 +6,10 @@ import { nodeSelected, nodeDeselected, removeNode, removeLink } from '../actions
 import { DiagramModel, LinkModel, ModelChangeEvent, ModelChangeEventType } from 'react-gojs';
 import { Action } from 'typescript-fsa';
 
-interface MyDiagramContainerStateProps {
+interface MyDiagramContainerProps {
     model: DiagramModel<NodeModel, LinkModel>;
+    onNodeSelection: (key: string, isSelected: boolean) => void;
```

Comment:
As MyDiagramContainer is a container component, it's a component "connected" to redux.
onNodeSelection and onModelChange are callbacks to dispatch redux actions. That is the reason why they are defined in  MyDiagramContainerDispatchProps interface.